pkg/ccl/backupccl/system_schema.go

```diff
@@ -113,6 +113,41 @@ func defaultSystemTableRestoreFunc(
 
 // Custom restore functions for different system tables.
 
+// tenantSettingsTableRestoreFunc restores the system.tenant_settings table. It
+// returns an error when trying to restore a non-empty tenant_settings table
+// into a non-system tenant.
+func tenantSettingsTableRestoreFunc(
+	ctx context.Context,
+	execCfg *sql.ExecutorConfig,
+	txn *kv.Txn,
+	systemTableName, tempTableName string,
+) error {
+	executor := execCfg.InternalExecutor
+
+	countQuery := fmt.Sprintf("SELECT count(1) FROM %s", tempTableName)
+	row, err := executor.QueryRow(ctx, "count-tenant-settings", txn, countQuery)
+	if err != nil {
+		return errors.Wrapf(err, "checking for existing tenant settings")
+	}
+
+	count, ok := row[0].(*tree.DInt)
+	if !ok {
+		return errors.AssertionFailedf("failed to read count as DInt (was %T)", count)
+	}
+
+	if int64(*count) > 0 && !execCfg.Codec.ForSystemTenant() {
```

Comment:
huh, can't we just... ignore them?

Reply:
Yes, I believe we can.  I've modified this to simply warn on a tenant restore with a non-empty tenant_settings table.  The warning might be a bit of overkill, but since this seems like the kind of assumption that could change in the future, it seemed like it might be useful for some future debugging.

Reply:
Fair enough -- we also discard the tenants themselves so I don't _particular_ worry about the tenant settings, but the log line and extra count() probably doesn't hurt.